src/checkbox.tsx

```diff
@@ -123,7 +123,13 @@ const Checkbox: React.FC<RenderProps | ChildrenProps> = (props) => {
             role="checkbox"
             aria-checked={value ?? checkedState}
             onKeyDown={disabled ? undefined : handleKeyDown}
-            onClick={disabled ? undefined : handleChange}
+            onClick={(e) => {
+                if (!disabled) {
+                    e.preventDefault();
```

Comment:
this preventDefault shouldn't be needed. There's no default handler in divs

Reply:
I'm unsure whether we should use the stopPropagation method on disabled fields or not. I would stopPropagation unconditionally. What's your opinion?

Reply:
It can be an option. I'm not sure what would be the best option from the user experience side. I will ask design what are their thoughts about this.

Reply:
hmmm... why would you ask design? This is just an implementation detail, isn't it?
I wouldn't add the handler if disabled

Reply:
Because it's related to user experience. Design agrees with Pedro about not propagating the event when it's disabled

Reply:
ok for me, then